factory/api/apiResolverFactory.go

```diff
@@ -413,8 +402,22 @@ func createScQueryElement(
 		MissingTrieNodesNotifier: syncer.NewMissingTrieNodesNotifier(),
 	}
 
+	var apiBlockchain data.ChainHandler
 	maxGasForVmQueries := args.generalConfig.VirtualMachine.GasConfig.ShardMaxGasPerVmQuery
 	if args.processComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		apiBlockchain, err = blockchain.NewMetaChain(disabled.NewAppStatusHandler())
```

Comment:
extract in 2 functions, so the code would be:
```
if shard == metachain:
    createMetaVmFactory(...)
else
    createShardVmFactory(...)
```

Reply:
done